tests/datatypes/test_str.py

```diff
@@ -301,57 +301,288 @@ def test_isupper(self):
-                if 'bee'.islower():
-                    print('bee is lowercase')
-                else:
-                    print('bee is not lowercase')
-
-                # Multiple character, mixed case
-                if 'Bee'.islower():
-                    print('Bee is lowercase')
-                else:
-                    print('Bee is not lowercase')
-
-                # Multiple character, all caps with digits and punctuation
-                if 'B1E2E!'.islower():
-                    print('B1E2E! is lowercase')
-                else:
-                    print('B1E2E! is not lowercase')
-
-                # Multiple character, mixed case with digits and punctuation
-                if 'B1e2E!'.islower():
-                    print('B1e2E! is lowercase')
-                else:
-                    print('B1e2E! is not lowercase')
-
-                # Only numbers and punctuation
-                if '12!3_4'.islower():
-                    print('12!3_4 is lowercase')
-                else:
-                    print('12!3_4 is not lowercase')
-
-                print('done.')
+    def test_islower(self):
+        self.assertCodeExecution("""
+            # Single capital
+            if 'B'.islower():
+                print('B is lowercase')
+            else:
+                print('B is not lowercase')
+
+            # Single lowercase
+            if 'b'.islower():
+                print('b is lowercase')
+            else:
+                print('b is not lowercase')
+
+            # Multiple character, all caps
+            if 'BEE'.islower():
+                print('BEE is lowercase')
+            else:
+                print('BEE is not lowercase')
+
+            # Multiple character, all lowercase
+            if 'bee'.islower():
+                print('bee is lowercase')
+            else:
+                print('bee is not lowercase')
+
+            # Multiple character, mixed case
+            if 'Bee'.islower():
+                print('Bee is lowercase')
+            else:
+                print('Bee is not lowercase')
+
+            # Multiple character, all caps with digits and punctuation
+            if 'B1E2E!'.islower():
+                print('B1E2E! is lowercase')
+            else:
+                print('B1E2E! is not lowercase')
+
+            # Multiple character, mixed case with digits and punctuation
+            if 'B1e2E!'.islower():
+                print('B1e2E! is lowercase')
+            else:
+                print('B1e2E! is not lowercase')
+
+            # Only numbers and punctuation
+            if '12!3_4'.islower():
+                print('12!3_4 is lowercase')
+            else:
+                print('12!3_4 is not lowercase')
+
+            print('done.')
+        """)
+
+    def test_lstrip(self):
+        # No argument passed, strip whitespace only
+        self.assertCodeExecution("""
+            x = ' Bee'
+            print(s.lstrip())
```

Comment:
I think you'll find this is a false positive test pass. The string is `x`, the test is on `s`. Both Batavia and Python will have the same result - but it will be a `NameError`, not the result of `s.lstrip()`.

On a related note - Batavia tests have a bit of an overhead associated with starting an `assertCodeExecution`, so it's advantageous to run a bunch of tests in one code block if possible (like `test_islower` does), rather than having a lot of independent test executions.